### PR TITLE
filters/auth: add login redirect stub support

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
For single page applications user may want to return javascript stub that stores location hash into localstorage and then redirects user to the authorization url.

### CODE REVIEW

File: docs/reference/filters.md [hunk 1]
```diff
@@ -1655,6 +1660,27 @@ all:
...
+                  localStorage.setItem('original-location-hash', window.location.hash);
+                }
+                window.location.replace('{{authCodeURL}}');
+              </script>
+            </head>
+          </html>
+    `)
+    -> oauthGrant()
+    -> "http://localhost:9090";
+```
```
Comment (szuecs) [inline docs/reference/filters.md:1682]: and now this inside a yaml :D

**Comment by szuecs** [comment]
:+1:

File: docs/reference/filters.md (hunk 1 above)
Comment (AlexanderYastrebov) [reply docs/reference/filters.md:1682]: Its fine actually:
```yaml
  routes:
    - pathSubtree: /
      filters:
        - |
          annotate("oauthGrant.loginRedirectStub", `
            <!doctype html>
            <html lang="en">
              <head>
                <title>Redirecting...</title>
                <script>
                  if (window.location.hash !== null) {
                    localStorage.setItem('original-location-hash', window.location.hash);
                  }
                  window.location.replace('{{authCodeURL}}');
                </script>
              </head>
            </html>
          `)
        - oauthGrant()
        - |
          inlineContent(`
            <!doctype html>
            <html lang="en">
              <head>
                <title>Hello world</title>
              </head>
              <body>
                <p id="hello">Hello world</p>

                <script>
                  const hash = localStorage.getItem('original-location-hash');
                  if (hash !== null) {
                    window.location.hash = hash;
                    localStorage.removeItem('original-location-hash');
                  }
                  document.getElementById('hello').innerText = 'Page: ' + window.location.hash;
                </script>
              </body>
            </html>
          `)
```

Comment (szuecs) [reply docs/reference/filters.md:1682]: :1st_place_medal:

**Comment by macedigital** [comment]
Wow, thanks for implementing this so quickly :rocket:

**Comment by macedigital** [comment]
:+1:

File: docs/reference/filters.md (hunk 1 above)
Comment (RomanZavodskikh) [reply docs/reference/filters.md:1682]: :exploding_head:

**Pull request merged**